src/layers/attention.h

```diff
@@ -504,12 +511,18 @@ class Attention {
 
     // score: M * K(keyLen), value: K * headSize, output: M * headSize
     template <typename T1, typename T2, typename T3>
-    void gemm2(T1 *score, T2 *value, T3 *output, int M, int headSize, int K, int lds, int ldv, int ldo) {
+    void gemm2(T1 *score, const std::tuple<T2 *, int, float *> &valueMat, T3 *output, int M, int headSize, int K,
```

Comment:
gemm2 -> gemm_score?

Reply:
will move to kernel in the progress, will get a better name then.